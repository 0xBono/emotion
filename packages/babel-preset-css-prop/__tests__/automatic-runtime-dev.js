import babelTester from 'babel-tester'
import preset from '@emotion/babel-preset-css-prop'

babelTester('automatic runtime dev', __dirname, {
  filename: '/test/path/to/some/file.js',
  presets: [
    [
      preset,
      {
        sourceMap: false,
        runtime: 'automatic',
        development: true
      }
    ]
  ]
})
