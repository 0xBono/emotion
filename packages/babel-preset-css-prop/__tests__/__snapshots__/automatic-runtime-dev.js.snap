// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automatic runtime dev array-css-prop 1`] = `
"import * as React from 'react'

const Component = props => <div css={[{ color: 'green' }]} {...props} />


      ↓ ↓ ↓ ↓ ↓ ↓

import { jsxDEV as _jsxDEV } from \\"@emotion/react/jsx-dev-runtime\\";
var _jsxFileName = \\"/test/path/to/some/file.js\\";
import * as React from 'react';
var _ref = {
  name: \\"v98kxt\\",
  styles: \\"color:green\\"
};

const Component = props => _jsxDEV(\\"div\\", {
  css: _ref,
  ...props
}, void 0, false, {
  fileName: _jsxFileName,
  lineNumber: 3,
  columnNumber: 28
}, this);"
`;

exports[`automatic runtime dev fragment 1`] = `
"export let Button = ({ loading, ...props }) => {
  return (
    <>
      <button css={{ color: 'hotpink' }} {...props} />
      {loading && <span>{'Loading...'}</span>}
    </>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import { jsxDEV as _jsxDEV } from \\"@emotion/react/jsx-dev-runtime\\";
import { Fragment as _Fragment } from \\"@emotion/react/jsx-dev-runtime\\";
var _jsxFileName = \\"/test/path/to/some/file.js\\";
var _ref = {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
};
export let Button = ({
  loading,
  ...props
}) => {
  return _jsxDEV(_Fragment, {
    children: [_jsxDEV(\\"button\\", {
      css: _ref,
      ...props
    }, void 0, false, {
      fileName: _jsxFileName,
      lineNumber: 4,
      columnNumber: 7
    }, this), loading && _jsxDEV(\\"span\\", {
      children: 'Loading...'
    }, void 0, false, {
      fileName: _jsxFileName,
      lineNumber: 5,
      columnNumber: 19
    }, this)]
  }, void 0, true);
};"
`;

exports[`automatic runtime dev index 1`] = `
"import * as React from 'react'

export let Button = props => {
  return (
    <>
      <button
        css={{
          color: 'hotpink'
        }}
        {...props}
      />
    </>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import { jsxDEV as _jsxDEV } from \\"@emotion/react/jsx-dev-runtime\\";
import { Fragment as _Fragment } from \\"@emotion/react/jsx-dev-runtime\\";
var _jsxFileName = \\"/test/path/to/some/file.js\\";
import * as React from 'react';
var _ref = {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
};
export let Button = props => {
  return _jsxDEV(_Fragment, {
    children: _jsxDEV(\\"button\\", {
      css: _ref,
      ...props
    }, void 0, false, {
      fileName: _jsxFileName,
      lineNumber: 6,
      columnNumber: 7
    }, this)
  }, void 0, false);
};"
`;

exports[`automatic runtime dev key-after-spread 1`] = `
"export let Buttons = ({ buttons }) => {
  return (
    <div>
      {buttons.map(({ id, label, ...rest }) => (
        <button
          {...rest}
          key={id}
          css={{
            color: 'hotpink'
          }}
        >
          {label}
        </button>
      ))}
    </div>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import { createElement as _createElement } from \\"@emotion/react\\";
import { jsxDEV as _jsxDEV } from \\"@emotion/react/jsx-dev-runtime\\";
var _jsxFileName = \\"/test/path/to/some/file.js\\";
var _ref = {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
};
export let Buttons = ({
  buttons
}) => {
  return _jsxDEV(\\"div\\", {
    children: buttons.map(({
      id,
      label,
      ...rest
    }) => _createElement(\\"button\\", { ...rest,
      key: id,
      css: _ref,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 5,
        columnNumber: 9
      },
      __self: this
    }, label))
  }, void 0, false, {
    fileName: _jsxFileName,
    lineNumber: 3,
    columnNumber: 5
  }, this);
};"
`;

exports[`automatic runtime dev no-static-children 1`] = `
"export let Button = props => {
  return (
    <button
      css={{
        color: 'hotpink'
      }}
      {...props}
    />
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import { jsxDEV as _jsxDEV } from \\"@emotion/react/jsx-dev-runtime\\";
var _jsxFileName = \\"/test/path/to/some/file.js\\";
var _ref = {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
};
export let Button = props => {
  return _jsxDEV(\\"button\\", {
    css: _ref,
    ...props
  }, void 0, false, {
    fileName: _jsxFileName,
    lineNumber: 3,
    columnNumber: 5
  }, this);
};"
`;

exports[`automatic runtime dev static-children 1`] = `
"export let Button = () => {
  return (
    <button
      css={{
        color: 'hotpink'
      }}
    >
      {'Test'}
    </button>
  )
}


      ↓ ↓ ↓ ↓ ↓ ↓

import { jsxDEV as _jsxDEV } from \\"@emotion/react/jsx-dev-runtime\\";
var _jsxFileName = \\"/test/path/to/some/file.js\\";
var _ref = {
  name: \\"3sn2xs\\",
  styles: \\"color:hotpink\\"
};
export let Button = () => {
  return _jsxDEV(\\"button\\", {
    css: _ref,
    children: 'Test'
  }, void 0, false, {
    fileName: _jsxFileName,
    lineNumber: 3,
    columnNumber: 5
  }, this);
};"
`;
